src/coreclr/src/jit/liveness.cpp

```diff
@@ -2477,17 +2477,16 @@ void Compiler::fgInterBlockLocalVarLiveness()
         if (VarSetOps::IsMember(this, exceptVars, varDsc->lvVarIndex) ||
             VarSetOps::IsMember(this, filterVars, varDsc->lvVarIndex))
         {
-            /* Mark the variable appropriately */
-            lvaSetVarDoNotEnregister(varNum DEBUGARG(DNER_LiveInOutOfHandler));
+            // Mark the variable appropriately.
+            lvaSetVarLiveInOutOfHandler(varNum);
         }
 
-        /* Mark all pointer variables live on exit from a 'finally'
-           block as either volatile for non-GC ref types or as
-           'explicitly initialized' (volatile and must-init) for GC-ref types */
+        // Mark all pointer variables live on exit from a 'finally' block as lvLiveInOutOfHndlr.
+        // and as 'explicitly initialized' (must-init) for GC-ref types.
 
         if (VarSetOps::IsMember(this, finallyVars, varDsc->lvVarIndex))
```

Comment:
I don't understand how this could be changed to `else`.
Also, note that the structure has not fundamentally changed - I just changed the comments to use `//` since I was modifying the code nearby.

Reply:
The previous condition is:
`if (smth1) lvaSetVarLiveInOutOfHandler(varNum);` 
and this is: 
`if (smth2) lvaSetVarLiveInOutOfHandler(varNum);`
, so we can do 
`if (smth1 or smth2) lvaSetVarLiveInOutOfHandler(varNum);` 
or
```
if (smth1) lvaSetVarLiveInOutOfHandler(varNum);
else if (smth2) lvaSetVarLiveInOutOfHandler(varNum);
```

As I see from `lvaSetVarLiveInOutOfHandler` we don't need to call it twice, do we?

Reply:
I see your point now. I'l restructure the conditions.